keps/sig-cli/2229-kubectl-xdg-base-dir/README

```diff
@@ -0,0 +1,235 @@
+  - [Graduation Criteria](#graduation-criteria)
+    - [Alpha -&gt; Beta Graduation](#alpha---beta-graduation)
+    - [Beta -&gt; GA Graduation](#beta---ga-graduation)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Implementation History](#implementation-history)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://github.com/kubernetes/enhancements/issues
+[kubernetes/kubernetes]: https://github.com/kubernetes/kubernetes
+[kubernetes/website]: https://github.com/kubernetes/website
+
+## Summary
+
+This enhancement is focused in provide all requirements for kubectl
+use the [XDG Base Directory Specification (XDG Spec)](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html).
+The XDG Spec have been used for a long time as the standard location to tools
+and programs store configuration files.
+
+## Motivation
+
+Today the kubectl store the configuration file via ~/.kube/ dir.
+However, there are multiple [community requests like #56402](https://github.com/kubernetes/kubernetes/issues/56402),
+from real users that have been looking for: 
+  - A single place for managing the configuration files
+  - Automation/Backup
+  - Use the same environment [variables as others projects](https://specifications.freedesktop.org/basedir-spec/latest/ar01s03.html)
+
+### Goals
+
+The goal is make kubectl follow the XDG Spec and automatic migrate
+the configuration from $HOME/.kube to $HOME/.config/kube wihout stopping
+kubectl to work.
+
+### Non-Goals
+
+Deprecate any file under $HOME/.kube/
+
+## Proposal
+
+- kubectl should follow the XDG Base Directory Specification
+- Use $HOME/.config/kube as default dir for configurations
+- Be compatible with $HOME/.kube until config migrated to $HOME/.config/kube
+- Smoothly migrate the config from $HOME/.kube to the new location
+- Update documentation related to ~/.kube
+- Write a blog post to explain and communicate such change
+
+### Risks and Mitigations
+
+Risks are limited, as none of the cluster components, will be affected
```

Comment:
There are many, many ecosystem tools, built on client-go or on other client libraries, that are going to assume the current location.

Couldn't this be mitigated for existing users by them just creating a symlink from ~/.kube -> ~/.config/kube? That would prevent breaking all these existing tools.

Reply:
Hi @johnbelamaric sure users can or even:

```
$ mkdir $HOME/.config/ && mv ~/.kube $HOME/.config/kube
$ echo "export KUBECONFIG=$HOME/.config/kube/config" >> ~/.bashrc
$ source ~/.bashrc
```

I am wrote the enhancement/PR as we had many issues opened on sig-cli about this topic (to support out of the box). However, if decided that's not the way to go, I am more than happy to close all issues (and this enhancement/PR) and suggest users to use export KUBECONFIG or manually create a symbol link.